x/gamm/keeper/pool_service.go

```diff
@@ -232,6 +236,11 @@ func (k Keeper) JoinSwapExternAmountIn(
 	k.hooks.AfterJoinPool(ctx, sender, pool.GetId(), addedCoins, shareOutAmount)
 	k.RecordTotalLiquidityIncrease(ctx, addedCoins)
 
+	err = k.UpdatePoolTwap(ctx, poolId, tokenIn.Denom)
```

Comment:
Can we make a helper function in keeper.go called `GetPoolAndPrepareForSwap`, and in that function run `k.GetPool()`, and `k.UpdatePoolTwap()`?

That will save us some code duplication within all these methods

Reply:
Err, it should be called `GetPoolAndPrepareForJoin` I think